### PR TITLE
Template app updates

### PR DESCRIPTION
**Pull request by MongoCaleb**
Add Atlas connection URL to each template app.

Reviewed? Please check the box:

[ X ]MAUI

[ X ]Swift

[x]Kotlin

[ X ]ReactNative

[  ]Flutter

### CODE REVIEW

File: sync-todo/v2/client/kotlin-sdk/app/src/main/res/values/atlasConfig.xml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <resources>
-  <string name="realm_app_id">sync-todo-v2-oxdgx</string>
+  <string name="realm_app_id">****</string>
```
Comment (cbullinger) [inline sync-todo/v2/client/kotlin-sdk/app/src/main/res/values/atlasConfig.xml:2]:
```suggestion
  <string name="realm_app_id">PUT YOUR APP ID HERE</string>
```

File: sync-todo/v2/client/kotlin-sdk/app/src/main/res/values/atlasConfig.xml [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
   <string name="realm_base_url">https://realm.mongodb.com</string>
+  <string name="dataExplorerLink">****</string>
```
Comment (cbullinger) [inline sync-todo/v2/client/kotlin-sdk/app/src/main/res/values/atlasConfig.xml:4]: it looks like the link is configured as appId + baseUrl + atlasUrl in the other SDKs -- should this be similarly constructed? Or a PUT_YOUR_DATA_EXPLORER_LINK_HERE placeholder if it's something users are expected to get?

Comment (MongoCaleb) [reply sync-todo/v2/client/kotlin-sdk/app/src/main/res/values/atlasConfig.xml:4]: This value is created when the app is created.

File: sync-todo/v2/client/swiftui/App/AppConfig.swift [hunk 3]
```diff
@@ -20,5 +21,15 @@ func loadAppConfig() -> AppConfig {
     let atlasConfigPropertyList = try! PropertyListSerialization.propertyList(from: data, format: nil) as! [String: Any]
     let appId = atlasConfigPropertyList["appId"]! as! String
     let baseUrl = atlasConfigPropertyList["baseUrl"]! as! String
-    return AppConfig(appId: appId, baseUrl: baseUrl)
+    
+    // If you're getting this app code by cloning the repository at
+    // https://github.com/mongodb/template-app-swiftui-todo
+    // it does not contain the data explorer link. Download the
+    // app template from the Atlas UI to view a link to your data.
```
Comment (cbullinger) [inline sync-todo/v2/client/swiftui/App/AppConfig.swift:28]: @MongoCaleb do we want to have something similar in the other apps?

File: sync-todo/v2/client/flutter/lib/main.dart [hunk 4]
```diff
@@ -8,13 +8,14 @@ import 'package:flutter_todo/realm/app_services.dart';
 import 'package:flutter_todo/screens/homepage.dart';
 import 'package:flutter_todo/screens/log_in.dart';
 
+dynamic realmConfig;
```
Comment (mongodben) [inline sync-todo/v2/client/flutter/lib/main.dart:11]: this seems funky. what's the thought for a top scope `dynamic` variable?

Comment (krollins-mdb) [reply sync-todo/v2/client/flutter/lib/main.dart:11]: This seems to be the easiest way to share the realmConfig info (including the Data Explorer URL) between `main()` and `App`. Do you have a suggestion for another way to do it?

**Comment by krollins-mdb** [comment]
@mongodben, would you mind taking another look at the Flutter changes? I've pushed some updates.

File: sync-todo/v2/client/flutter/lib/components/todo_list.dart [hunk 5]
```diff
@@ -27,6 +30,10 @@ class _TodoListState extends State<TodoList> {
   @override
   Widget build(BuildContext context) {
     final realmServices = Provider.of<RealmServices>(context);
+    final atlasUrl = widget.atlasUrl;
+
+    print("To see your data in Atlas, follow this link:$atlasUrl");
```
Comment (mongodben) [inline sync-todo/v2/client/flutter/lib/components/todo_list.dart:35]:
```suggestion
    print("To see your data in Atlas, follow this link: $atlasUrl");
```

File: sync-todo/v2/client/flutter/lib/components/todo_list.dart [hunk 6]
```diff
@@ -92,4 +116,14 @@ class _TodoListState extends State<TodoList> {
       ],
     );
   }
+
+  launchURL({required String url}) async {
```
Comment (mongodben) [inline sync-todo/v2/client/flutter/lib/components/todo_list.dart:120]:
```suggestion
  _launchURL({required String url}) async {
```

make private method.

also personally, i would make this a private func that's scoped to file level so it's more testable (even if you aren't actually writing tests for it)

File: sync-todo/v2/client/flutter/lib/main.dart (hunk 4 above)
Comment (mongodben) [reply sync-todo/v2/client/flutter/lib/main.dart:11]: can you pass it as a variable from main to App?

**Review by mongodben — CHANGES_REQUESTED** [review]
couple small things

File: sync-todo/v2/client/flutter/lib/components/todo_list.dart (hunk 6 above)
Comment (krollins-mdb) [reply sync-todo/v2/client/flutter/lib/components/todo_list.dart:120]: Oops. Yeah, that's a great point. Thank you!

File: sync-todo/v2/client/flutter/lib/main.dart (hunk 4 above)
Comment (krollins-mdb) [reply sync-todo/v2/client/flutter/lib/main.dart]: 🙃 Of course! lol

I've been context switching too much. Updated to remove the global dynamic.

File: sync-todo/v2/client/flutter/lib/main.dart [hunk 7]
```diff
@@ -31,18 +34,48 @@ class App extends StatelessWidget {
...
+        },
       ),
     );
   }
 }
 
+// This class gets app info from `atlasConfig.json`, which is
+// populated with field by the server when you download the
+// template app through the Atlas App Services UI or CLI.
+class Config extends ChangeNotifier {
```
[the remark this answers is not in the document]
Comment (mongodben) [reply sync-todo/v2/client/flutter/lib/main.dart:62]: this approach of adding the class makes sense to me.

consider having `assets/config/atlasConfig.json` be passed as an arg to the static method:

smthn like:

```dart
 static Future<Config> getConfig(jsonConfigPath: string) async {
    dynamic realmConfig = json
        .decode(await rootBundle.loadString(jsonConfigPath));

    var config = Config._create(realmConfig);

    return config;
  }
```

and then pass that string at your app level.

doesn't really matter here, but better application design

**Review by mongodben — CHANGES_REQUESTED** [review]
small back to you

File: sync-todo/v2/client/flutter/lib/main.dart (hunk 7 above)
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply sync-todo/v2/client/flutter/lib/main.dart:62]: Great point! And easy to do. 😄

**Review by mongodben — APPROVED** [review]
nice work @krollins-mdb. lgtm!

**Pull request merged**